### PR TITLE
Fixing a couple of issues with typescript builds.

### PR DESCRIPTION
**Pull request by dzearing**

### CODE REVIEW

File: src/TypeScriptTask.ts [hunk 1]
```diff
@@ -41,12 +42,18 @@ export class TypeScriptTask extends GulpTask<ITypeScriptTaskConfig> {
     ],
     reporter: {
       error: (error: ITypeScriptErrorObject) => {
+        let errorMessage = error.diagnostic.messageText;
```
Comment (nickpape-msft) [inline src/TypeScriptTask.ts]: could you just say

const errorMessage: string = (typeof errorMessage === 'object' ? (errorMessage as { messageText: string }).messageText : errorMessage);

Then you don't have to constantly cast messageText to a string

Comment (dzearing) [inline src/TypeScriptTask.ts]: Sure. I've updated the PR.

File: src/TypeScriptTask.ts [hunk 2]
```diff
@@ -42,18 +42,16 @@ export class TypeScriptTask extends GulpTask<ITypeScriptTaskConfig> {
...
-        let errorMessage = error.diagnostic.messageText;
-
-        if (typeof errorMessage === 'object') {
-          errorMessage = (errorMessage as { messageText: string }).messageText as string;
-        }
+        const errorMessage: string = (typeof error.diagnostic.messageText === 'object') ?
+          (error.diagnostic.messageText as { messageText: string }).messageText :
+          error.diagnostic.messageText as string;
 
         this.fileError(
```
Comment (dzearing) [inline src/TypeScriptTask.ts]: @nickpape-msft look better?

**Comment by harry8525** [comment]
Approved

**Pull request merged**